pkg/util/strategicpatch/patch.go

```diff
@@ -52,7 +53,12 @@ func StrategicMergePatchData(original, patch []byte, dataStruct interface{}) ([]
 		return nil, fmt.Errorf("strategic merge patch needs a struct, %s received instead", t.Kind().String())
 	}
 
-	result, err := mergeMap(o, p, t)
+	diffWriter := &diffWriter{
```

Comment:
You're altering `StrategicMergePatchData` to always print a diff to stdout, which is not desirable behavior for all callers. It would be better to factor out the contained logic into an implementation func that takes an optional diffWriter. That way StrategicMergePatchData can use a dummy diffWriter that doesn't write anything, and StrategicMergeDfif can pass an actual diffWriter and use the result.